specs/_features/whisk/beacon-chain.md

```diff
@@ -446,13 +412,16 @@ def apply_deposit(state: BeaconState,
         # Initialize validator if the deposit signature is valid
         if bls.Verify(pubkey, signing_root, signature):
             index = get_index_for_new_validator(state)
-            validator = get_validator_from_deposit_whisk(state, pubkey, withdrawal_credentials, amount)
+            validator = get_validator_from_deposit(pubkey, withdrawal_credentials, amount)
             set_or_append_list(state.validators, index, validator)
             set_or_append_list(state.balances, index, amount)
-            # [New in Altair]
             set_or_append_list(state.previous_epoch_participation, index, ParticipationFlags(0b0000_0000))
             set_or_append_list(state.current_epoch_participation, index, ParticipationFlags(0b0000_0000))
             set_or_append_list(state.inactivity_scores, index, uint64(0))
+            # [New in Whisk]
+            k = get_unique_whisk_k(state, ValidatorIndex(len(state.validators) - 1))
```

Comment:
This was changed to `len(state.validators)-1`. Was the previous code off-by-one buggy?

Reply:
Ah, makes sense!